README.md

```diff
@@ -185,6 +185,44 @@ In combination with a `SerialName` specified for the child class, you have full
 }
 ```
 
+<h4>Serialization of Updates</h4>
+Firestore contains update methods that allow for multiple fields to be updated at the same time. 
+This sdk offers special update methods that allow for applying custom serialization to each individual field though an update builder.
+Where an `update` method exists, an `updateFields` method will also be available. In this, each value can have its serializer customized:
+
+```kotlin
+documentRef.updateFields {
+    "field" to "value"
+    // Set the value of otherField to "1" using a custom Serializer
+    "otherField".to(IntAsStringSerializer(), 1)
+    
+    // Overwrite build settings. All fields added after this will have these build settings applied
+    buildSettings = {
```

Comment:
Good point. I've refactored it to a method called 'encodeNextWith'.

Usage:

```kotlin
documentRef.updateFields {
    "field" to "value"
    // Set the value of otherField to "1" using a custom Serializer
    "otherField".to(IntAsStringSerializer(), 1)
    
    // Overwrite build settings. All fields added after this will have these build settings applied
    encodeNextWith {
        encodeDefaults = true
        serializersModule = module
    }
    "city" to abstractCity
}
```

Reply:
why not just add encodeWith as an optional arg to the `to` function?

Reply:
Because then you'd have to add the same block to every call to 'to':

```
"field".to("value") { encodeDefaults = true }
"otherField".to("otherValue") { encodeDefaults = true }
```

Just looks less clean in my opinion.

Reply:
yes thats not great either. But the `blockThatAffectsEverythingBelowIt {}` pattern doesn't feel that intuitive to me.

What about `updateFields` taking the encode settings as optional arguments in addition to being able to pass them into the to function, so it would become:

```
documentRef.updateFields(encodeDefaults = true) {
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    
    "city".to(abstractCity) { serializersModule = module }
}
```
Although that does beg the question why not just have the encode settings as optional arguments in the `to` function also, making it simply:

```
documentRef.updateFields(encodeDefaults = true) {
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    
    "city".to(abstractCity, serializersModule = module)
}
```

thoughts?

Reply:
Ok, finally had the time to look at it.

I went for a different approach: I made these DSL classes extend the EncodeSettings.Builder and made them defer actually encoding the data until the end. This allows for the introduction of a `withEncodeSettings` block, that creates a nested builder to overwrite the settings. Usage:

```kotlin
documentRef.updateFields(encodeDefaults = true) {
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    
    withEncodeSettings {
        serializersModule = module
        "city".to(abstractCity)
    }
}
```

Reply:
Note that now, you can set the encodeSettings anywhere in the block:

```
serializersModule = module
"city" to abstractCity
```

should be the same as
```
"city" to abstractCity
serializersModule = module
```

Reply:
so `withEncodeSettings` is just to restrict the settings to certain properties otherwise you can just set the encodeSettings at the root level? e.g:

```
documentRef.updateFields {
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    encodeDefaults = true
}
```

Reply:
Correct. You can set the encodeSettings directly in the root dsl as well. The `withEncodeSettings` block allows for setting custom settings that apply only to the values added within the block. Seems like the nicest way to me. Its flexible and relatively readable.

Reply:
Yeah it's much better - I even think it's not necessary to also accept the settings as optional arguments to `updateFields`, so:
```
documentRef.updateFields(encodeDefaults = true) {
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    
    withEncodeSettings {
        serializersModule = module
        "city" to abstractCity
    }
}
```
becomes:
```
documentRef.updateFields {
    encodeDefaults = true
    "field" to "value"
    "otherField".to(IntAsStringSerializer(), 1)
    
    withEncodeSettings {
        serializersModule = module
        "city" to abstractCity
    }
}
````

Reply:
can you update the readme with that latest syntax?